scripts/utils/create_vm.sh

```diff
@@ -1,93 +1,57 @@
 #!/bin/bash
-set -e
-echo "Creating vm"
 
-if [ "$#" -ne 4 ]; then
-    echo "Illegal number of parameters"
-fi
+echo "Creating vm"
 
 ARCHITECTURE=$1
 CHANNEL=$2
-USE_PROXY=$3
-PORT=$4
+PORT=$3
+BRANCH=$4
+
+export WORKSPACE=${WORKSPACE:-$(pwd)}
 
 # Define variables
-WORKSPACE=$(pwd)
 if [ $ARCHITECTURE == "amd64" ]; then
 	ASSERTION=nested-amd64.model
     QEMU=qemu-system-x86_64
+    PLATFORM=pc-amd64
+    IMG="pc.img"
 elif [ $ARCHITECTURE == "i386" ]; then
 	ASSERTION=nested-i386.model
     QEMU=qemu-system-i386
+    PLATFORM=pc-i386
+    IMG="pc.img"
 else
 	echo "Architecture $ARCHITECTURE not supported"
 	exit 1
 fi
 
-# Configure the proxy if needed
-if [ $USE_PROXY == "USE_PROXY" ]; then
-    export http_proxy=http://squid.internal:3128
-    export HTTP_PROXY=http://squid.internal:3128
-    export https_proxy=https://squid.internal:3128
-    export HTTPS_PROXY=https://squid.internal:3128
-fi
-
 # Install the dependencies
-sudo apt install -y snapd qemu genisoimage sshpass
-sudo apt install -y ubuntu-image
-sudo apt install -y unzip
-sudo apt install -y sshpass
+sudo apt install -y snapd qemu genisoimage sshpass unzip
+sudo snap install ubuntu-image --edge --classic
 
 # Prepare the image under test
 WORKDIR=$WORKSPACE/work-dir
 mkdir -p $WORKDIR
 
-git clone https://github.com/snapcore/snapd
-sudo ubuntu-image --image-size 3G snapd/tests/lib/assertions/$ASSERTION --channel $CHANNEL --output ubuntu-core.img
-mv ubuntu-core.img $WORKDIR
-
+if [ ! -d snapd ]; then 
+    git clone https://github.com/snapcore/snapd
+fi
+sudo /snap/bin/ubuntu-image --image-size 3G snapd/tests/lib/assertions/$ASSERTION --channel $CHANNEL -O $WORKDIR
```

Comment:
As we talked online, using an image created with a model signed by a non-canonical key can lead to bad results, @pedronis can probably shed some light on this